src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -113,13 +113,15 @@ type Db = MVar (BlockEnv SQLiteEnv)
 doRestore :: ChainwebVersion -> ChainId -> Db -> Maybe (BlockHeight, ParentHash) -> IO PactDbEnv'
 doRestore v cid dbenv (Just (bh, hash)) = runBlockEnv dbenv $ do
     setModuleNameFix
+    setSortedKeys
     clearPendingTxState
     void $ withSavepoint PreBlock $ handlePossibleRewind v cid bh hash
     beginSavepoint Block
     return $! PactDbEnv' $! PactDbEnv chainwebPactDb dbenv
   where
     -- Module name fix follows the restore call to checkpointer.
-    setModuleNameFix = modify' $ set bsModuleNameFix $! enableModuleNameFix v bh
+    setModuleNameFix = bsModuleNameFix .= enableModuleNameFix v bh
+    setSortedKeys = bsSortedKeys .= pact420Upgrade v bh
```

Comment:
Does this matter much given both are sequenced monadically? Could also just make the field strict in the data type.

Reply:
As an aside, it would have been better to have a single call to `modify' $ set bsModuleNameFix foo . set bsSorted bar` to avoid the extra roundtrip to the monad. But it's once per tx so I'm sure the load is insignificant.